### ISSUE TITLE
generator not found

### ISSUE THREAD

**Issue by FusionJed** [mistricky/codesnap.nvim#129]
I'm new to nvim... well, to programming in general so i don't know what's broken

```
rustc --version
rustc 1.81.0 (eeb90cda1 2024-09-04)
```

```
cargo --version
cargo 1.81.0 (2dbb1af80 2024-08-20)
```

platform - x86_64-unknown-linux-gnu
Config: 
```
return {
  "mistricky/codesnap.nvim",
  build = "make build_generator",
  keys = {
    { "<leader>cc", "<cmd>CodeSnap<cr>", mode = "x", desc = "Save selected code snapshot into clipboard" },
    { "<leader>cs", "<cmd>CodeSnapSave<cr>", mode = "x", desc = "Save selected code snapshot in ~/Pictures" },
  },
  opts = {
    save_path = "~/Pictures",
    has_breadcrumbs = true,
    bg_theme = "bamboo",
  },
}

```

Issue: 

![image](https://github.com/user-attachments/assets/7c480472-dd53-497f-9712-3ad007b06e1e)

**Comment by realhackcraft** [comment]
I’m experiencing the same issue. The commands `build = “make build_generator”`, `build = “make”`, and `build = “make generator”` all fail to execute.

**Comment by mistricky** [comment]
Hi @realhackcraft @FusionJed  thx for using CodeSnap, I could not reproduce this issue on my computer, I'm on Apple Silicon also. CodeSnap has cross-compile the `generator` on the following three platforms:

- x86_64-unknown-linux-gnu
- x86_64-apple-darwin
- aarch64-apple-darwin

`x86_64-unknown-linux-gnu` and `aarch64-apple-darwin` is bundled in source code, so typically you guys just need run `make` before install, the CodeSnap will install on your computer seamlessly, for instance `Lazy.nvim`:

```
build = "make"
```
> Refer to https://github.com/mistricky/codesnap.nvim?tab=readme-ov-file#install

But generally you can also set build to `make build_generator` to build CodeSnap from source code (for this case @realhackcraft there is no such `make generator` command), if this build process not successful, please post the fail log.

You can try the following steps to reinstall CodeSnap:
- Change build to `make`
- Open Lazy.nvim board
- Cleanup CodeSnap
- Restart Neovim

**Comment by FusionJed** [comment]
Hello, It seems I installed CodeSnap without _rust_ resulting in missing files. As of right now, I installed rust and reinstalled (Cleaned up CodeSnap in :Lazy) and it's working flawlessly.

![image](https://github.com/user-attachments/assets/a69dab3b-9276-45ae-9e31-a7d26a7331e8)

I tried reproducing the issue and the issue was i installed CodeSnap _without_ rust

resulting with the following errors:

![image](https://github.com/user-attachments/assets/7e1839ee-7ca7-4ccc-8cdb-3668834f342a)

![image](https://github.com/user-attachments/assets/19c89f18-757c-4f3d-bd70-627a247161d2)

Thank you for taking time and responding
CodeSnap is Great : D

**Comment by realhackcraft** [comment]
It also worked for me when I reinstalled it, although the reason is more mysterious. I already have rust installed on my system and I had installed CodeSnap a few months ago, and then uninstalled it. This week, when I tried to install the plugin again, the build failed.

It could be because I didn't put the build="make" in the plugin when I first installed it, and then after I added that, it didn't run the build because it thought it was already installed.
